containers/daap-python-base/src/var/task/data_platform_paths.py

```diff
@@ -0,0 +1,219 @@
+from __future__ import annotations
+from datetime import datetime
+
+import os
+import re
+from dataclasses import dataclass
+from typing import NamedTuple
+import uuid
+
+
+EXTRACTION_TIMESTAMP_FORMAT = "%Y%m%dT%H%M%SZ"
+EXTRACTION_TIMESTAMP_REGEX = re.compile(
+    r"^(.*)/(extraction_timestamp=)([0-9TZ]{1,16})/(.*)$"
+)
+
+
+class BucketPath(NamedTuple):
+    """
+    A path to an object in S3
+    """
+
+    bucket: str
+    prefix: str
```

Comment:
in AWS they have a bucket and a key. is there any advantage in having prefix here rather than just carrying these concepts over from aws and calling this key?

Reply:
Nope, I was just confused about the correct AWS terminology here. I will change it to key.

Reply:
I can see sometimes you have a directory name in the prefix field, this does sometimes happen in s3 if you create folders in the s3 console but I think officially a key always relates to a specific file. I guess I'm trying to say that prefix might still be a valid concept in this class, but I can't think of a use case for it atm.